src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -664,42 +631,31 @@ private boolean hasSameNameAsSuperClass(String superClassName, TypeDeclDesc type
     private static TypeDeclDesc getTheNearestClass(String outerTypeDeclName,
             List<TypeDeclDesc> typeDeclWithSameName) {
         return Collections.min(typeDeclWithSameName, (first, second) -> {
-            int diff = Integer.compare(
-                    typeDeclNameMatchingCount(outerTypeDeclName, second.getQualifiedName()),
-                    typeDeclNameMatchingCount(outerTypeDeclName, first.getQualifiedName()));
-            if (diff == 0) {
-                diff = Integer.compare(first.getDepth(), second.getDepth());
-            }
-            return diff;
+            return getTypeDeclarationNameMatchingCountDiff(outerTypeDeclName, first, second);
         });
     }
 
     /**
-     * Calculates and returns the type declaration name matching count.
+     * Get the difference between type declaration name matching count. If the
+     * difference between them is zero, then their depth is compared to obtain the result.
      *
-     * <p>
-     * Suppose our pattern class is {@code foo.a.b} and class to be matched is
-     * {@code foo.a.ball} then type declaration name matching count would be calculated by
-     * comparing every character, and updating main counter when we hit "." to prevent matching
-     * "a.b" with "a.ball". In this case type declaration name matching count
-     * would be equal to 6 and not 7 (b of ball is not counted).
-     * </p>
-     * Duplicated, until <a>https://github.com/checkstyle/checkstyle/issues/11201</a>
-     *
-     * @param patternClass class against which the given class has to be matched
-     * @param classToBeMatched class to be matched
-     * @return class name matching count
+     * @param outerTypeDeclName outer type declaration of anonymous inner class
+     * @param firstTypeDecl first input type declaration
+     * @param secondTypeDecl second input type declaration
+     * @return difference between type declaration name matching count
      */
-    private static int typeDeclNameMatchingCount(String patternClass, String classToBeMatched) {
-        final char packageSeparator = PACKAGE_SEPARATOR.charAt(0);
-        final int length = Math.min(classToBeMatched.length(), patternClass.length());
-        int result = 0;
-        for (int i = 0; i < length && patternClass.charAt(i) == classToBeMatched.charAt(i); ++i) {
-            if (patternClass.charAt(i) == packageSeparator) {
-                result = i;
-            }
+    private static int getTypeDeclarationNameMatchingCountDiff(String outerTypeDeclName,
+                                                               TypeDeclDesc firstTypeDecl,
+                                                               TypeDeclDesc secondTypeDecl) {
```

Comment:
This method was required as 
`CheckUtil.typeDeclarationNameMatchingCount(outerTypeDeclName, second.getQualifiedName())`
was exceeding line length and wrapping it on the next line resulted in exceeding max lambda length (10).